### PR TITLE
Python 3.4/5/6 and PyPy

### LINKED ISSUE #1: Drop support for 2.6, add support for 3.4/3.5/PyPy

### PR DESCRIPTION
**Pull request by jamadden**
- Stop claiming/testing support for Python 2.6 and 3.3.
- Modern .travis.yml/tox.ini
- 100% coverage and enable coveralls
- Rename .txt -> .rst.
- Test and fix complex slices in CompositeQueue.

Fixes #1

### CODE REVIEW

File: src/zc/queue/_queue.py [hunk 1]
```diff
@@ -235,32 +235,23 @@ def __iter__(self):
...
-                if ix < start:
-                    continue
-                stride_ct -= 1
-                if stride_ct == 0:
-                    res.append(v)
-                    stride_ct = stride
-            return res
+            # We do this the easy way instead of attempting to handle
+            # negative strides ourself
+            return list(self)[index]
```
Comment (mgedmin) [inline src/zc/queue/_queue.py:242]: I think you could preserve the laziness by using `return list(itertools.islice(self, *index.indices(len(self))))`.

And then I look at those four closing parens and think maybe that's not such a good idea after all.

Maybe `return list(itertools.islice(self, index.start, index.stop, index.step))`?  That does look better.

**Review by mgedmin — APPROVED** [review]
LGTM

File: src/zc/queue/_queue.py (hunk 1 above)
Comment (jamadden) [reply src/zc/queue/_queue.py:242]: I don't see it [explicitly in the documentation](https://docs.python.org/2.7/library/itertools.html#itertools.islice) but `islice` doesn't handle negative steps:
```pycon
>>> import itertools
>>> itertools.islice((), None, None, -1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Step for islice() must be a positive integer or None.
```

Comment (jamadden) [reply src/zc/queue/_queue.py:242]: Is it worth having two branches?
```python
if isinstance(index, slice):
   if index.step is not None and index.step > 0:
      return list(itertools.islice(self, index.start, index.stop, index.step))
   return list(self)[index] # negative steps aren't handled by islice, it doesn't buffer
```

Comment (jamadden) [reply src/zc/queue/_queue.py:242]: Oh, wait, there's the docs. I skipped right over that sentence:

> Unlike regular slicing, islice() does not support negative values for start, stop, or step.

So it's actually pretty limited.

Comment (mgedmin) [reply src/zc/queue/_queue.py:242]: Never mind then!

**Comment by jamadden** [comment]
Thanks! A PyPI release would be nice, but it isn't blocking anything.

**Comment by jamadden** [comment]
> Package Index Owner: gary, zvezdan, sshyub, J1m, srichter

Pinging @jimfulton and @strichter (sorry I'm not sure how the other accounts map.)

(Our PyPI usernames are `mgedmin` and `jamadden` if you're too busy to make a release and don't mind other people doing it.)

**Comment by jimfulton** [comment]
On Thu, May 11, 2017 at 9:18 AM, Jason Madden <notifications@github.com>
wrote:
...

> (Our PyPI usernames are mgedmin and jamadden if you're too busy to make a
> release and don't mind other people doing it.)
>
> You're both owners now.

**Comment by jamadden** [comment]
Thank you!

**Pull request merged**